src/miral/window_info.cpp

```diff
@@ -114,11 +116,15 @@ miral::WindowInfo::Self::Self(Window window, WindowSpecification const& params)
     min_aspect(optional_value_or_default(params.min_aspect(), default_min_aspect_ratio)),
     max_aspect(optional_value_or_default(params.max_aspect(), default_max_aspect_ratio)),
     shell_chrome(optional_value_or_default(params.shell_chrome(), mir_shell_chrome_normal)),
-    depth_layer(optional_value_or_default(params.depth_layer(), mir_depth_layer_application))
+    depth_layer(optional_value_or_default(params.depth_layer(), mir_depth_layer_application)),
+    attached_edges(optional_value_or_default(params.attached_edges(), mir_placement_gravity_center))
```

Comment:
It isn't obvious why this should default to "center". Are we using that as a synonym for "none"?

Reply:
Attaching to the center of the output seems like as good a default as any. I don't think it's terribly important what the default is, as the value will generally be set at the same time as the state is changed to attached.

Reply:
Is it *mandatory* that the attachment should be set when switching to attached state? That sounds reasonable?

Reply:
No. If it was then that would imply it's mandatory to set the edges every time you switch into an attached state, even if you had already set the edges the previous time. Also, that would also imply it's impossible to change the attached edges when the window is not in an attached state, which in some cases may be convenient. I don't see the potential for problems with just having them be independent properties.

Reply:
As `attached_edges` is always set to _something_ then not requiring it to be set when switching to "attached" is consistent. If that's what we elect to do then any arbitrary default value works and "center" is the canonical "middle of the road".